src/Rings/MPolyMap/AffineAlgebras.jl

```diff
@@ -5,9 +5,10 @@
 ################################################################################
 
 const AffAlgHom = MPolyAnyMap{DT, CT, Nothing} where {T <: FieldElem,
-                                                       U <: MPolyRingElem{T},
-                                                       DT <: Union{MPolyRing{T}, MPolyQuoRing{U}},
-                                                       CT <: Union{MPolyRing{T}, MPolyQuoRing{U}}}
+                                                       U1 <: MPolyRingElem{T},
+                                                       U2 <: MPolyRingElem{T}, # types in domain an codomain might differ: one might be decorated, the other not.
+                                                       DT <: Union{MPolyRing{T}, MPolyQuoRing{U1}},
+                                                       CT <: Union{MPolyRing{T}, MPolyQuoRing{U2}}}
```

Comment:
```suggestion
                                                       DT <: Union{MPolyRing{T}, MPolyQuoRing{<:MPolyRingElem{T}}},
                                                       CT <: Union{MPolyRing{T}, MPolyQuoRing{<:MPolyRingElem{T}}}}
```
For the case of different Us, there is no need to specify them concretely.

Reply:
Yes, seems reasonable. I just preferred to stay close to what I found from before, also to not confuse the original authors unnecessarily much in case they want to look and confirm. Either way is fine with me, but personally I wouldn't bother to change this.